osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -138,14 +217,71 @@ public void handleGetFirmwareVersionResponse(final List<FirmwareVersionDto> firm
 
         this.checkFirmwareHistory(ids.getDeviceIdentification(), firmwareVersions);
 
-        final ResponseMessage responseMessage = ResponseMessage.newResponseMessageBuilder()
-                .withIds(ids)
-                .withResult(result)
-                .withOsgpException(osgpException)
-                .withDataObject((Serializable) firmwareVersions)
-                .withMessagePriority(messagePriority)
-                .build();
-        this.webServiceResponseMessageSender.send(responseMessage);
+        final boolean isPendingFirmwareUpdate = this.checkSsldPendingFirmwareUpdate(ids, firmwareVersions);
+
+        if (!isPendingFirmwareUpdate) {
+            final ResponseMessage responseMessage = ResponseMessage.newResponseMessageBuilder()
+                    .withIds(ids)
+                    .withResult(result)
+                    .withOsgpException(osgpException)
+                    .withDataObject((Serializable) firmwareVersions)
+                    .withMessagePriority(messagePriority)
+                    .build();
+            this.webServiceResponseMessageSender.send(responseMessage);
+        }
+    }
+
+    private boolean checkSsldPendingFirmwareUpdate(final CorrelationIds ids,
+            final List<FirmwareVersion> firmwareVersions) {
+
+        final String deviceIdentification = ids.getDeviceIdentification();
+        final String organisationIdentification = ids.getOrganisationIdentification();
+        final String correlationUid = ids.getCorrelationUid();
+
+        final Ssld ssld = this.ssldRepository.findByDeviceIdentification(deviceIdentification);
+        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = ssld.getSsldPendingFirmwareUpdate();
+        if (ssldPendingFirmwareUpdate == null) {
+            return false;
+        }
+        if (!organisationIdentification.equals(ssldPendingFirmwareUpdate.getOrganisationIdentification())) {
```

Comment:
Are the checks on organization and correlation id needed? Would it be sufficient to check on device identification (as long as this check is only called after receiving a device registration message from osgp core)?